### ISSUE TITLE
[fix] time picker component should always have a label

### ISSUE THREAD

**Issue by shkeating** [trussworks/react-uswds#2611]
**ReactUSWDS Version: 5.4.0 & USWDS Version: 3.6.0**

# Describe the bug

In our storybook examples of the Combo box component, we do not always include a label. 

In the USWDS docs about the time picker, it is explicitly called out that we must **always use a label** on this component. We should not include examples where it is unlabelled in our project storybook. The following examples lack a label:

- [Default Time Picker](https://trussworks.github.io/react-uswds/?path=/story/components-form-controls-time-picker--default-time-picker)

We could simply remove the example that shows the time picker with no label from storybook, or in this case, it may be helpful to set the label prop in this component to be required to help guide react-uswds users to ensure they don't forget it in their implementations. I could be convinced of either approach, as both have pros and cons. This could be potentially cumbersome to projects already using react-uswds that have already added labels of their own who later upgrade versions.

# To Reproduce

1. Go to [Default Time Picker](https://trussworks.github.io/react-uswds/?path=/story/components-form-controls-time-picker--default-time-picker) 
2. See there is no label

# Expected behavior

All visible versions of the time picker in our storybook should have labels to prevent confusion that they are optional, when they are mandatory for this component. 

# Additional context

- [USWDS Time Picker documentation](https://designsystem.digital.gov/components/time-picker/)
- [Labelling Controls - W3C](https://www.w3.org/WAI/tutorials/forms/labels/)
- file(s) to be edited: `src/components/forms/TimePicker/TimePicker.tsx` and/or `src/components/forms/TimePicker/TimePicker.stories.tsx`

# Screenshots

<img width="703" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/c004626b-0d3e-455d-bd1b-447a54164ea9">
